numpy/array_api/_array_object.py

```diff
@@ -108,6 +108,17 @@ def __repr__(self: Array, /) -> str:
             mid = np.array2string(self._array, separator=', ', prefix=prefix, suffix=suffix)
         return prefix + mid + suffix
 
+    # This function is not required by the spec, but we implement it here for
+    # convenience so that np.asarray(np.array_api.Array) will work.
+    def __array__(self, dtype=None):
```

Comment:
```suggestion
    def __array__(self, dtype: None | np.dtype[Any] = None) -> npt.NDArray[Any]:
```
I don't think `import numpy.typing as npt` is already present in the file, so it will have to be added.
You could consider adding it the `if TYPE_CHECKING` block, considering none of the `npt` members are used here during runtime anyway.